test/functional/blockchain.py

```diff
@@ -44,6 +45,36 @@ def run_test(self):
         self._test_stopatheight()
         assert self.nodes[0].verifychain(4, 0)
 
+    def _test_getblockchaininfo(self):
+        self.log.info("Test getblockchaininfo")
+
+        keys = [
+            'bestblockhash',
+            'bip9_softforks',
+            'blocks',
+            'chain',
+            'chainwork',
+            'difficulty',
+            'headers',
+            'mediantime',
+            'pruned',
+            'softforks',
+            'verificationprogress',
+        ]
+        res = self.nodes[0].getblockchaininfo()
+        # result should have pruneheight and default keys if pruning is enabled
+        assert_equal(sorted(res.keys()), sorted(['pruneheight'] + keys))
+        # pruneheight should be greater or equal to 0
+        assert res['pruneheight'] >= 0
+
+        # restart node
+        self.extra_args[0] = ['-stopatheight=207']
```

Comment:
Sorry, I misled you here. You should use `self.nodes[0].extra_args = ...`. The `self.extra_args` class variable is only used when instantiating the TestNode. That's important for `restart_node()`